Respect TORCH_DISABLE_ADDR2LINE in symbolizer

zdevito opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #121359

If TORCH_DISABLE_ADDR2LINE is set, the symbolizer will instead give the
filename of the shared library as the filename, the offset in that library as the linenumber,
and use dladdr to get the function name if possible. This is much faster than using addr2line,
and the symbols can be later resolved offline using addr2line if desired.

zdevito committed:
Respect TORCH_DISABLE_ADDR2LINE in symbolizer

If TORCH_DISABLE_ADDR2LINE is set, the symbolizer will instead give the
filename of the shared library as the filename, the offset in that library as the linenumber,
and use dladdr to get the function name if possible. This is much faster than using addr2line,
and the symbols can be later resolved offline using addr2line if desired.

[ghstack-poisoned]

zdevito committed:
Update on "Respect TORCH_DISABLE_ADDR2LINE in symbolizer"


If TORCH_DISABLE_ADDR2LINE is set, the symbolizer will instead give the
filename of the shared library as the filename, the offset in that library as the linenumber,
and use dladdr to get the function name if possible. This is much faster than using addr2line,
and the symbols can be later resolved offline using addr2line if desired.

[ghstack-poisoned]

aaronenyeshi reviewed (approved):
Great fallback solution! Performing addr2line for later make sense. Please check that linter errors get first before landing.

zdevito committed:
Update on "Respect TORCH_DISABLE_ADDR2LINE in symbolizer"


If TORCH_DISABLE_ADDR2LINE is set, the symbolizer will instead give the
filename of the shared library as the filename, the offset in that library as the linenumber,
and use dladdr to get the function name if possible. This is much faster than using addr2line,
and the symbols can be later resolved offline using addr2line if desired.

[ghstack-poisoned]

zdevito commented:
@pytorchbot merge

zdevito commented:
@pytorchbot merge
